controller/appcontroller.go

```diff
@@ -657,7 +673,11 @@ func (ctrl *ApplicationController) processAppRefreshQueueItem() (processNext boo
 		localManifests = opState.Operation.Sync.Manifests
 	}
 
-	compareResult, err := ctrl.appStateManager.CompareAppState(app, "", app.Spec.Source, refreshType == appv1.RefreshTypeHard, localManifests)
+	revision := ""
+	if comparisonLevel == ComparisonLevelRecent {
```

Comment:
would it be clearer to set this to `app.Spec.Souce.TargetRevision`. Elsewhere we assume `""` means `HEAD`.

Reply:
Agree, done.